src/glue/jobs/compare_parquet_datasets.py

```diff
@@ -0,0 +1,473 @@
+            f"{data_type}: Staging dataset has the following missing columns:\n{str(missing_cols)}"
+        )
+    if add_cols:
+        compare_msg.append(
+            f"{data_type}: Staging dataset has the following additional columns:\n{str(add_cols)}"
+        )
+    return compare_msg
+
+
+def get_data_types_to_compare(
+    s3: boto3.client, bucket_name: str, staging_namespace: str, main_namespace: str
+) -> list:
+    """This gets the common data types to run the comparison of the parquet datasets from
+    the two namespaced paths on based on the folders in the s3 bucket"""
+    staging_datatype_folders = get_folders_in_s3_bucket(
+        s3, bucket_name, namespace=staging_namespace
+    )
+    main_datatype_folders = get_folders_in_s3_bucket(
+        s3, bucket_name, namespace=main_namespace
+    )
+    return list(set(staging_datatype_folders) & set(main_datatype_folders))
+
+
+def compare_dataset_data_types(
+    s3: boto3.client, bucket_name: str, staging_namespace: str, main_namespace: str
+) -> list:
+    """This looks at the current datatype folders in the S3 bucket between the
+    two namespaced paths and outputs a message if there are any differences
+    in the datatype folders"""
+    compare_msg = []
+    staging_datatype_folders = get_folders_in_s3_bucket(
+        s3, bucket_name, namespace=staging_namespace
+    )
+    main_datatype_folders = get_folders_in_s3_bucket(
+        s3, bucket_name, namespace=main_namespace
+    )
+    missing_datatypes = list(set(main_datatype_folders) - set(staging_datatype_folders))
+    add_datatypes = list(set(staging_datatype_folders) - set(main_datatype_folders))
+
+    if missing_datatypes:
+        compare_msg.append(
+            f"Staging dataset has the following missing data types: {str(missing_datatypes)}"
+        )
+
+    if add_datatypes:
+        compare_msg.append(
+            f"Staging dataset has the following additional data types: {str(add_datatypes)}"
+        )
+    return compare_msg
+
+
+def compare_datasets_and_output_report(
+    data_type: str,
+    staging_dataset: pd.DataFrame,
+    main_dataset: pd.DataFrame,
+    staging_namespace: str,
+    main_namespace: str,
+) -> str:
+    """This method prints out a human-readable report summarizing and
+    sampling differences between datasets for the given data type.
+    A full list of comparisons can be found in the datacompy package site.
+
+    Returns:
+        str: large string block of the report
+    """
+    compare = datacompy.Compare(
+        df1=staging_dataset,
+        df2=main_dataset,
+        join_columns=INDEX_FIELD_MAP[data_type],
+        abs_tol=0,  # Optional, defaults to 0
+        rel_tol=0,  # Optional, defaults to 0
+        df1_name=staging_namespace,  # Optional, defaults to 'df1'
+        df2_name=main_namespace,  # Optional, defaults to 'df2'
+    )
+    compare.matches(ignore_extra_columns=False)
+    return compare.report()
+
+
+def add_additional_msg_to_comparison_report(
```

Comment:
All the string formatting messages scattered about are not pretty/ideal but tried to group/limit them to just this function and `compare_datasets_by_data_type`.

Reply:
I think this is great!